sqlmesh/core/state_sync/base.py

```diff
@@ -138,11 +138,15 @@ def get_environments(self) -> t.List[Environment]:
         """
 
     @abc.abstractmethod
-    def max_interval_end_for_environment(self, environment: str) -> t.Optional[int]:
+    def max_interval_end_for_environment(
+        self, environment: str, model_fqns: t.Optional[t.Set[str]] = None
```

Comment:
i don't think we need to use "fqn" everywhere, there's really only one representation now, could just call it models, the type sig is a str, but whatever up to you